src/tools/rbd/Utils.cc

```diff
@@ -442,9 +442,11 @@ int get_pool_image_snapshot_names(const po::variables_map &vm,
   //Validate pool name while creating/renaming/copying/cloning/importing/etc
   if (spec_validation == SPEC_VALIDATION_FULL) {
     boost::regex pattern("^[^@/]+?$");
-    if (!boost::regex_match (*pool_name, pattern)) {
-      std::cerr << "rbd: invalid pool name '" << *pool_name << "'" << std::endl;
-      return -EINVAL;
+    if (pool_name != nullptr && pool_name->empty()) {
```

Comment:
Why are you checking that the ```pool_name``` is empty?

Reply:
@dillaman Thanks for review.
since pool_name is never checked for NULL in function before. Either we should check pool_name to be nonnull initially and return.
But before passing pool_name to regex_match() API we should check its nonnull.
Coverity raised the Issue, Please have a look at Coverity ID:1394846

Reply:
That wasn't my question -- I understand and agree w/ the nullptr check (even though in practice it will never be hit). I asked why did you also insert ```&& pool_name->empty()``` since if the pool name is empty, what's the point of validating the reg ex?